capsules/src/net/icmpv6/icmpv6.rs

```diff
@@ -0,0 +1,182 @@
+//! ICMPv6 layer of the Tock networking stack.
+//!
+//! - Author: Conor McAvity <cmcavity@stanford.edu>
+
+use net::stream::{decode_u16, decode_u32, decode_u8};
+use net::stream::{encode_u16, encode_u32, encode_u8};
+use net::stream::SResult;
+
+#[derive(Copy, Clone)]
+pub struct ICMP6Header {
+    pub code: u8,
+    pub cksum: u16,
+    pub options: ICMP6HeaderOptions,
+    pub len: u16, // Not ICMP field, here for convenience
+}
+
+#[derive(Copy, Clone)]
+pub enum ICMP6HeaderOptions {
+    Type1 { unused: u32 },
+    Type3 { unused: u32 },
+    Type128 { id: u16, seqno: u16 },
+    Type129 { id: u16, seqno: u16 },
+}
+
+#[derive(Copy, Clone)]
+pub enum ICMP6Type {
```

Comment:
This type seems redundant to me since the caller can just ignore the data in `ICMP6HeaderOptions`